scripts/core/ui/components/content-create-dropdown/more-templates.tsx

```diff
@@ -156,45 +61,50 @@ export class MoreTemplates extends React.PureComponent<IProps, IState> {
                         />
                         <span className="form-label" style={{minHeight: 0}}>{gettext('More templates')}</span>
                     </Spacer>
-
                     <SpacerBlock v gap="4" />
-
                     <Input
                         type="text"
                         labelHidden
-                        value={state.searchString}
+                        inlineLabel
+                        value={this.state.searchString}
                         onChange={(val) => {
-                            const nextState: IState = {
-                                ...state,
-                                searchString: val,
-                            };
-
-                            this.setState(nextState);
+                            this.setState({searchString: val});
                         }}
                     />
-
                     <div className="content-create-dropdown--spacer" />
                 </div>
-
-                <ResizeObserverComponent>
-                    {({width}) => {
-                        if (state.loading === true) {
-                            return null;
-                        } else {
-                            return (
-                                <VirtualList
-                                    width={width}
-                                    height={this.props.height - this.headerEl.offsetHeight}
-                                    totalItemsCount={state.templatesCount}
-                                    initialItems={state.templatesData}
-                                    loadItems={this.loadMoreItems}
-                                    itemTemplate={this.itemTemplate}
-                                    getId={(item) => item._id}
-                                />
-                            );
+                {
+                    <WithPagination
+                        getItems={(pageNo, pageSize, signal) => this.fetchData(pageNo, pageSize, signal)
+                            .then((res) => Promise.resolve({items: res._items, itemCount: res._meta.total}))
```

Comment:
no need for `Promise.resolve` on this line. You can return the object directly and return type doesn't change because of this